### PR TITLE
Mac/Linux shell scripts!

### PR DESCRIPTION
**Pull request by Raelifin**
I'm a little confused about runTestCases. The output I get is
```
Testing Assignment
Exit Code is 42
Testing BoolFunction
Exit Code is 0
Testing BreakContinue
Exit Code is 0
Testing Class
Exit Code is 0
Testing CodePointFunction
Exit Code is 0
Testing Comments
Exit Code is 0
Testing DoWhile
Exit Code is 0
Testing ElementAccess
Exit Code is 0
Testing ElseIf
Exit Code is 0
Testing Empty
Exit Code is 0
Testing EnumStruct
Exit Code is 0
Testing Equality
Exit Code is 0
Testing EscapeSequences
Exit Code is 0
Testing For
Exit Code is 0
Testing Function
Exit Code is 1
Testing FunctionOrder
Exit Code is 1
Testing Function_OneArg
Exit Code is 42
Testing Function_TwoArgs
Exit Code is 42
Testing GlobalLet
Exit Code is 42
Testing GlobalVar
Exit Code is 42
Testing HelloWorld
Hello World!
Exit Code is 0
Testing If
Exit Code is 0
Testing IfElse
Exit Code is 0
Testing List
Hello
HelloWorld!
sb[0]: value
sb[1]: value
Exit Code is 0
Testing LocalLet
Exit Code is 42
Testing LocalVar
Exit Code is 42
Testing LogicalOperators
Exit Code is 0
Testing Loop
Exit Code is 0
Testing Mutable
Hello
Exit Code is 0
Testing NewString
Exit Code is 0
Testing NewStringBuilder
Exit Code is 0
Testing Nullable
Exit Code is 0
Testing Plus
Exit Code is 0
Testing RelationalOperators
Exit Code is 0
Testing Resources
Hello to
all my "cool" friends\enemies.
Goodbye world!
Exit Code is 0
Testing Return0
Exit Code is 0
Testing Return42
Exit Code is 42
Testing UnaryMinus
Exit Code is 0
Testing UnsignedInt
Exit Code is 0
Testing while
Exit Code is 0
```

It looks like some of them are failing, but I'm not sure. Also, what does `enabledelayedexpansion` do?

### CODE REVIEW

**Comment by WalkerCodeRanger** [comment]
This addresses issue #4 "Create shell scripts for building on mac/linux".

**Comment by WalkerCodeRanger** [comment]
I haven't checked every result, but those test run results seem reasonable.  There are many test programs that intentionally return an error code.  The problem is that 0 would be the default return, so to verify that you are actually returning something, you have to return non-zero.  Eventually, I'd like to have a test harness that knows the expected exit code and reports only mismatches.

enabledelayedexpansion is a batch file work around for loops.  Notice that there are some variables surrounded by exclamation points instead of percents.  Those are delayed expansion variables.  Otherwise the command running expands the variable too early before the value of the variable is set.  I think the unix shell just does the right thing.

File: build/nix/runTestCases.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# This script builds and runs the test cases expected values
+./buildTestCases.sh
+
+# TODO What is this??
+# setlocal enabledelayedexpansion
```
Comment (WalkerCodeRanger) [inline build/nix/runTestCases.sh]: Not needed on unix, just delete

File: build/nix/buildCurrent.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+mkdir -p ../../translated/current
+rm -rf ../../translated/current/*
+source_files=`ls -d ../../src/*.ad ../../src/**/*.ad`
+# Use the previous version of the compiler to compile the new version into translated/current
+../../target/previous/Program $source_files -o ../../translated/current/Program.cpp -r ../../src/RuntimeLibrary.cpp -r ../../src/RuntimeLibrary.h
+
+# Build the new program into target/current for testing
+mkdir -p ../../target/current
+rm -rf ../../target/current/*
+clang++ ../../translated/current/*.cpp -o ../../target/current/Program -std=c++14 -Xclang -flto-visibility-public-std
```
Comment (WalkerCodeRanger) [inline build/nix/buildCurrent.sh]: remove `-Xclang -flto-visibility-public-std` if it doesn't cause the build to break.  Pretty sure this is a windows specific issue  (note this in other scripts as well)

**Review by WalkerCodeRanger — CHANGES_REQUESTED** [review]

**Comment by Raelifin** [comment]
Removed the Windows-specific code, as suggested

**Review by WalkerCodeRanger — APPROVED** [review]

**Pull request merged**